brave/brave-core
# fix(wallet): Disable Shield Button if an Account is already Shielded

## Description 

Disables the `Switch to Shielded Account` button if an `Account` in the keyring has already been `Shielded`


Resolves <https://github.com/brave/brave-browser/issues/42294>



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
1. Make sure that ZCash Shielded Transactions` is enabled
2. Open the `Wallet` and navigate to the `Accounts` tab
3. Create 2 `ZCash` accounts
4. Click on the `More Menu` for one of your ZCash Accounts
5. Click on `Switch to shielded account`
6. Now click `Shield account`
7. After the `Account` is shielded, open the `More Menu` for your other `ZCash` account
8. You should not see a `Switch to shielded account` button.

https://github.com/user-attachments/assets/df5a25ac-91e5-4c3c-8c31-045dcd146100
